pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -886,26 +873,50 @@ func getServiceNameFromContainerTags(tags []string) (string, string) {
 	return "", ""
 }
 
-func (s *discovery) enrichContainerData(service *model.Service, containers map[string]*agentPayload.Container, pidToCid map[int]string) {
-	id, ok := pidToCid[service.PID]
-	if !ok {
-		return
+func (s *discovery) getContainersMap() map[int]*workloadmeta.Container {
+	containers := s.wmeta.ListContainers()
+	containersMap := make(map[int]*workloadmeta.Container)
+	for _, container := range containers {
+		containersMap[int(container.PID)] = container
 	}
+	return containersMap
+}
 
-	container, ok := containers[id]
+func (s *discovery) getProcessContainerInfo(pid int, containers map[int]*workloadmeta.Container) (string, []string, error) {
+	container, ok := containers[pid]
 	if !ok {
+		return "", nil, fmt.Errorf("container not found for pid %d", pid)
+	}
+
+	containerID := container.EntityID.ID
+	collectorTags := container.CollectorTags
+
+	entityID := types.NewEntityID(types.ContainerID, containerID)
+	entityTags, err := s.tagger.Tag(entityID, types.HighCardinality)
```

Comment:
Maybe add a comment to this code saying it's borrowed from `GetContainers`? Otherwise it's not obvious why it's done this way (using CollectorTags, passing HighCardinality, etc).